ui/src/views/Hooks/ListHooks/index.jsx

```diff
@@ -46,25 +54,74 @@ export default class ListHooks extends Component {
     });
   };
 
+  updateExpanded() {
+    const query = parse(window.location.search.slice(1));
+    const {
+      data: { hookGroups },
+    } = this.props;
+    const { search } = query;
+    const expandedHookGroups = hookGroups
+      ? hookGroups
+          .filter(
+            group =>
+              group.hooks.filter(hook => hook.hookId.includes(search))
+                .length !== 0
+          )
+          .map(group => group.hookGroupId)
+      : [];
+
+    this.setState({ expanded: expandedHookGroups, selected: [] });
+  }
+
+  componentDidMount() {
+    this.updateExpanded();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.location.search.slice(1) !== this.props.location.search.slice(1)
+    ) {
+      this.updateExpanded();
+    }
+  }
+
   render() {
     const {
       classes,
       description,
       data: { loading, error, hookGroups },
     } = this.props;
-    const query = qs.parse(this.props.location.search.slice(1));
-    const hookSearch = query.search;
+    const { search } = parse(window.location.search.slice(1));
     const tree = hookGroups
-      ? hookGroups.map(group => ({
-          value: group.hookGroupId,
-          nodes: group.hooks.map(hook => ({
-            value: hook.hookId,
-            href: `/hooks/${group.hookGroupId}/${encodeURIComponent(
-              hook.hookId
-            )}`,
-          })),
-        }))
+      ? hookGroups.map(group => {
+          const nodes = group.hooks.map(hook => (
```

Comment:
I would move this whole block into a separate component and file so when you pass the hook into this new component as a prop, its only going to re-render everything in that component if the prop has changed. Right now, if a user selects or expands (state change), this is going to be rendered even if the hookGroups hasn't changed.

Reply:
can we have that as a follow up ticket? I would be adding a generic treeview component instead, this one is limited to depth 2 so it might be confusing, alternatively I could just move it as a functional component on top of the same file so we achieve short term rendering performance, what do you think?

Reply:
Moving it to the top of the file is probably fine.